utils/python/CIME/test_scheduler.py

```diff
@@ -80,7 +80,7 @@ def __init__(self, test_names, test_data=None,
             self._test_root = self._test_root.replace("$PROJECT", self._project)
 
         self._test_root = os.path.abspath(self._test_root)
-        self._test_id   = test_id if test_id is not None else CIME.utils.get_timestamp()
+        self._test_id   = test_id
```

Comment:
I think I might be using test_scheduler via library call from some of the scripts_regression_tests. I probably provide a test_id, but as long as the API is what is it, there's the chance of test_id being None.